File: CS/SnapSeparatorExample/Form1.cs
#region #Reference
using System;
using System.Data;
using System.Windows.Forms;
using DevExpress.Snap.Core.API;
using DevExpress.XtraRichEdit.API.Native;
// ...
#endregion #Reference

namespace SnapSeparatorExample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        #region #Code
        private void Form1_Load(object sender, EventArgs e) {
            // Create a dataset and bind the Snap control to data.
            DataSet ds = new DataSet();
            DataTable dt = new DataTable("Customers");
            dt.Columns.Add("Id", typeof(int));
            dt.Columns.Add("Name", typeof(string));
            dt.Rows.Add(1, "Steven Buchanan");
            dt.Rows.Add(2, "Anne Dodsworth");
            dt.Rows.Add(3, "Janet Levering");
            ds.Tables.Add(dt);
            this.snapControl1.DataSources.Add("DS", ds);

            // Create a Snap list and populate it with the data.
            SnapList snList = this.snapControl1.Document.CreateSnList(
                this.snapControl1.Document.Range.Start, "List1");
            snList.BeginUpdate();
            snList.DataSourceName = "DS";
            snList.DataMember = "Customers";
            SnapDocument listRow = snList.RowTemplate;
            Table listRowTable = listRow.Tables.Create(listRow.Range.Start, 1, 2);
            TableCellCollection listRowCells = listRowTable.FirstRow.Cells;
            listRow.CreateSnText(listRowCells[0].ContentRange.End, @"Id");
            listRow.CreateSnText(listRowCells[1].ContentRange.End, @"Name");

            // Define a list separator and specify its format.
            snList.Separator.AppendText(new string('\f', 1));

            // Finalize the Snap list creation.
            snList.EndUpdate();
            snList.Field.Update();
        }
        #endregion #Code
    }
}

File: VB/SnapSeparatorExample/Form1.vb
'#Region "#Reference"
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports DevExpress.Snap.Core.API
Imports DevExpress.XtraRichEdit.API.Native

' ...
'#End Region  ' #Reference
Namespace SnapSeparatorExample

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

'#Region "#Code"
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            ' Create a dataset and bind the Snap control to data.
            Dim ds As DataSet = New DataSet()
            Dim dt As DataTable = New DataTable("Customers")
            dt.Columns.Add("Id", GetType(Integer))
            dt.Columns.Add("Name", GetType(String))
            dt.Rows.Add(1, "Steven Buchanan")
            dt.Rows.Add(2, "Anne Dodsworth")
            dt.Rows.Add(3, "Janet Levering")
            ds.Tables.Add(dt)
            snapControl1.DataSources.Add("DS", ds)
            ' Create a Snap list and populate it with the data.
            Dim snList As SnapList = snapControl1.Document.CreateSnList(snapControl1.Document.Range.Start, "List1")
            snList.BeginUpdate()
            snList.DataSourceName = "DS"
            snList.DataMember = "Customers"
            Dim listRow As SnapDocument = snList.RowTemplate
            Dim listRowTable As Table = listRow.Tables.Create(listRow.Range.Start, 1, 2)
            Dim listRowCells As TableCellCollection = listRowTable.FirstRow.Cells
            listRow.CreateSnText(listRowCells(0).ContentRange.End, "Id")
            listRow.CreateSnText(listRowCells(1).ContentRange.End, "Name")
            ' Define a list separator and specify its format.
            snList.Separator.AppendText(New String(Microsoft.VisualBasic.Strings.ChrW(12), 1))
            ' Finalize the Snap list creation.
            snList.EndUpdate()
            snList.Field.Update()
        End Sub
'#End Region  ' #Code
    End Class
End Namespace
